datalake/src/main/java/com/sequenceiq/datalake/flow/create/SdxCreateActions.java

```diff
@@ -230,6 +245,14 @@ protected void doExecute(SdxContext context, SdxCreateFailedEvent payload, Map<O
                 } catch (NotFoundException notFoundException) {
                     LOGGER.info("Can not set status to SDX_CREATION_FAILED because data lake was not found");
                 }
+                // When SDX is created as part of RESIZE flow chain, we should mark flow as failed to avoid
+                // subsequent flow's to be executed.
+                flowLogService.getLastFlowLog(context.getFlowParameters().getFlowId()).ifPresent(flowLog -> {
+                    if (flowLog.getFlowChainId() != null) {
+                        Flow flow = getFlow(context.getFlowParameters().getFlowId());
+                        flow.setFlowFailed(payload.getException());
+                    }
+                });
```

Comment:
marking flow failed can cause various issues, for example user may can click retry on it because of failed flow, also flow API will return with failed flow, but the creation actually not failed in this case, so I recommend other approach for this, because marking any flow failed without reason can be confusing

Reply:
This piece of code is in "SDX_CREATION_FAILED_STATE" state. When the flow reaches this state, it is clear that the creation of SDX failed. I do not understand why you think creation flow enters this state without an actual failure?


Moreover, this logic is hit only when the create flow is part of a flow chain. This is will avoid any regression on current sdx creation.

Reply:
ohh, I haven't seen it is a FAILED state, then it should be ok, I think, sorry